### PR TITLE
API Upgrade to behat 3

### PR DESCRIPTION
**Pull request by tractorcow**
See https://github.com/silverstripe/silverstripe-behat-extension/pull/152

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -26,17 +26,17 @@
...
     "vlucas/phpdotenv": "^2.4"
   },
   "require-dev": {
     "phpunit/phpunit": "^5.7",
     "silverstripe/versioned": "^1.0@dev",
-    "silverstripe/behat-extension": "^2.1.0",
+    "silverstripe/behat-extension": "^3",
     "silverstripe/serve": "dev-master",
-    "silverstripe/testsession": "^2.0.0-alpha3",
+    "silverstripe/testsession": "^3",
```
Comment (sminnee) [inline composer.json]: Do we need to explicitly declare this dependency or can we just let behat-extension pull it in? Less is more :-)

Comment (tractorcow) [inline composer.json]: Let's remove it and let behat control this.

**Comment by sminnee** [comment]
Behat 3.3 is now installed, but it fails with the following error:

```                                                                                                                             
  [Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException]                                               
  The service "silverstripe_extension.context.initializer" has a dependency on a non-existent parameter "behat.paths.base".  
```

File: behat.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  extensions:
+    SilverStripe\BehatExtension\MinkExtension:
+      default_session: selenium2
+      javascript_session: selenium2
+      selenium2:
+        browser: firefox
+
+    SilverStripe\BehatExtension\Extension:
+      screenshot_path: %behat.paths.base%/tests/behat/artifacts/screenshots
```
Comment (sminnee) [inline behat.yml]: Looks like this should be `%paths.base%` now

**Comment by sminnee** [comment]
next error:

```
[Behat\Testwork\Suite\Exception\SuiteConfigurationException]        
  Suite configuration for a suite "framework" is already registered.  
```

File: behat.yml (hunk 2 above)
Comment (tractorcow) [inline behat.yml]: Yes correct.

**Comment by sminnee** [comment]
```
PHP Fatal error:  Class 'SilverStripe\CMS\Model\SiteTree' not found in /home/travis/build/silverstripe/silverstripe-framework/vendor/silverstripe/behat-extension/src/Context/SilverStripeContext.php on line 271
PHP Fatal error:  Uncaught exception 'Behat\Testwork\Call\Exception\CallErrorException' with message 'Notice: Undefined index: SERVER_PROTOCOL in /home/travis/build/silverstripe/silverstripe-framework/src/Control/HTTPResponse.php line 290' in /home/travis/build/silverstripe/silverstripe-framework/vendor/silverstripe/behat-extension/src/Utility/RetryableCallHandler.php:94
Stack trace:
#0 /home/travis/build/silverstripe/silverstripe-framework/src/Control/HTTPResponse.php(290): SilverStripe\BehatExtension\Utility\RetryableCallHandler->handleError(8, 'Undefined index...', '/home/travis/bu...', 290, Array)
#1 /home/travis/build/silverstripe/silverstripe-framework/src/Logging/HTTPOutputHandler.php(144): SilverStripe\Control\HTTPResponse->output()
#2 /home/travis/build/silverstripe/silverstripe-framework/vendor/monolog/monolog/src/Monolog/Handler/AbstractProcessingHandler.php(37): SilverStripe\Logging\HTTPOutputHandler->write(Array)
#3 /home/travis/build/silverstripe/silverstripe-framework/vendor/monolog/monolog/src/Monolog/Logger.ph in /home/travis/build/silverstripe/silverstripe-framework/vendor/silverstripe/behat-extension/src/Utility/RetryableCallHandler.php on line 94
Fatal error: Class 'SilverStripe\CMS\Model\SiteTree' not found in /home/travis/build/silverstripe/silverstripe-framework/vendor/silverstripe/behat-extension/src/Context/SilverStripeContext.php on line 271
```

**Comment by tractorcow** [comment]
This can be merged if you want to then move onto getting CMS behat tests passing.

**Comment by sminnee** [comment]
I was reviewing the CMS tests first to see how close to passing I could get it.

**Pull request merged**